.github/workflows/build-test-publish-images.yml

```diff
@@ -0,0 +1,246 @@
+            NOTEBOOKS_TAG_PREFIX="rapidsai-notebooks-${pr_num}-"
+          fi
+          echo "BASE_TAG_PREFIX=${BASE_TAG_PREFIX}" | tee -a ${GITHUB_OUTPUT}
+          echo "NOTEBOOKS_TAG_PREFIX=${NOTEBOOKS_TAG_PREFIX}" | tee -a ${GITHUB_OUTPUT}
+      - name: Compute image repo
+        id: compute-image-repo
+        run: |
+          base_repo="rapidsai"
+          notebooks_repo="rapidsai-notebooks"
+          if [ "${{ inputs.build_type }}" = "pull-request" ]; then
+            base_repo="staging"
+            notebooks_repo="staging"
+          fi
+          echo "BASE_IMAGE_REPO=${base_repo}" | tee -a ${GITHUB_OUTPUT}
+          echo "NOTEBOOKS_IMAGE_REPO=${notebooks_repo}" | tee -a ${GITHUB_OUTPUT}
+      - name: Compute RAPIDS_VER
+        id: compute-rapids-ver
+        run: |
+          GIT_DESCRIBE_TAG="$(git describe --tags --abbrev=0)"
+          GIT_DESCRIBE_TAG="${GIT_DESCRIBE_TAG:1}" # remove leading 'v'
+          ALPHA_TAG=""
+          if [[ $GIT_DESCRIBE_TAG =~ [a-z] ]]; then
+            rapids-logger "Most recent tag is an alpha tag"
+            ALPHA_TAG="a"
+          fi
+          RAPIDS_VER="$(echo $GIT_DESCRIBE_TAG | awk 'BEGIN{FS=OFS="."} NF--')" # Convert full tag to YY.MM
+
+          echo "RAPIDS_VER=${RAPIDS_VER}" | tee -a ${GITHUB_OUTPUT}
+          echo "ALPHA_TAG=${ALPHA_TAG}" | tee -a ${GITHUB_OUTPUT}
+      - name: Compute test matrix
+        id: compute-test-matrix
+        run: |
+          TEST_MATRIX=$(yq '.${{ inputs.build_type }}' matrix-test.yaml)
+          export TEST_MATRIX
+
+          echo "TEST_MATRIX=$(yq -n -o json 'env(TEST_MATRIX)' | jq -c '{include: .}')" | tee --append "${GITHUB_OUTPUT}"
+  build:
+    needs: compute-matrix
+    strategy:
+      matrix: ${{ fromJSON(needs.compute-matrix.outputs.MATRIX) }}
+      fail-fast: false
+    secrets: inherit
+    uses: ./.github/workflows/build-image.yml
+    with:
+      ARCHES: ${{ toJSON(matrix.ARCHES) }}
+      CUDA_VER: ${{ matrix.CUDA_VER }}
+      LINUX_VER: ${{ matrix.LINUX_VER }}
+      PYTHON_VER: ${{ matrix.PYTHON_VER }}
+      RAPIDS_VER: ${{ needs.compute-matrix.outputs.RAPIDS_VER }}
+      DASK_SQL_VER: ${{ matrix.DASK_SQL_VER }}
+      BASE_TAG:
+        "rapidsai/${{ needs.compute-matrix.outputs.BASE_IMAGE_REPO }}:\
+        ${{ needs.compute-matrix.outputs.BASE_TAG_PREFIX }}\
+        ${{ needs.compute-matrix.outputs.RAPIDS_VER }}\
+        ${{ needs.compute-matrix.outputs.ALPHA_TAG }}-\
+        cuda${{ matrix.CUDA_VER }}-\
+        py${{ matrix.PYTHON_VER }}"
+      NOTEBOOKS_TAG:
+        "rapidsai/${{ needs.compute-matrix.outputs.NOTEBOOKS_IMAGE_REPO }}:\
+        ${{ needs.compute-matrix.outputs.NOTEBOOKS_TAG_PREFIX }}\
+        ${{ needs.compute-matrix.outputs.RAPIDS_VER }}\
+        ${{ needs.compute-matrix.outputs.ALPHA_TAG }}-\
+        cuda${{ matrix.CUDA_VER }}-\
+        py${{ matrix.PYTHON_VER }}"
+  test:
+    needs: [build, compute-matrix]
+    # TODO: nightly tests
```

Comment:
IIUC right now we only have `pull-request` (pr.yml, which builds and tests) and `branch` (publish.yml, which builds and publishes). So in addition to updating this bit of this workflow, we'll also need to add an additional new workflow for nightly tests. Should we track that work in a new issue?

Reply:
Yes, there are a several follow ups to this PR that I intend to open issues and/or PRs for.